app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,89 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    @task = Task.find_by(id: params[:id])
+    if @task.nil?
+      redirect_to tasks_path
+      return
+    end
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.new(task_params)
+    if @task.save
+      redirect_to "/tasks/#{Task.last.id}"
```

Comment:
Instead of interpolating the string you can just pass `@task` into the `task_path` method:

```suggestion
      redirect_to task_path(@task)
```